dialogs.c

```diff
@@ -338,6 +345,27 @@ static struct iio_context * get_context(Dialogs *data)
 			return get_context_from_osc();
 		}
 		return ctx;
+#ifdef SERIAL_BACKEND
+	} else if (gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(dialogs.connect_serial))) {
+		struct iio_context *ctx;
+		gchar *port = gtk_combo_box_get_active_text(
+				GTK_COMBO_BOX(dialogs.connect_seriald));
+		gchar *baud_rate = gtk_combo_box_get_active_text(
+				GTK_COMBO_BOX(dialogs.connect_serialbr));
+
+		/* Size is +3: for ':', ',' and '\0' */
+		int buf_size = sizeof("serial:") + strlen(port) + strlen(baud_rate) + 3;
+		gchar *result = malloc(buf_size);
+		snprintf(result, buf_size, "serial:%s,%s", port, baud_rate);
```

Comment:
Please change this to [g_strdup_printf()](https://developer.gnome.org/glib/stable/glib-String-Utility-Functions.html#g-strdup-printf) in a follow up patch.